packages/next/install-swc.js

```diff
@@ -0,0 +1,256 @@
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+const tar = require('next/dist/compiled/tar')
+
+if (process.env.NEXT_SWC_SKIP_INSTALL) {
+  process.exit(0)
+}
+
+const MAX_VERSIONS_TO_CACHE = 5
+const { version } = require('next/package.json')
+
+// @ts-ignore https://nodejs.org/api/report.html
+const { glibcVersionRuntime } = process.report.getReport().header
+const isGlibc = !!glibcVersionRuntime
+
+/**
+  @type {
+  [platform: string]: {
+    [arch: string]: {
+      // function to check if the arch/platform fully matches
+      // e.g. check if linux is glibc/musl
+      check?: () => {}
+      packageName: string
+    }[]
+  }
+}
+*/
+const packageMap = {
+  win32: {
+    x64: [
+      {
+        packageName: '@next/swc-win32-x64-msvc',
+      },
+    ],
+    ia32: [
+      {
+        packageName: '@next/swc-win32-ia32-msvc',
+      },
+    ],
+    arm64: [
+      {
+        packageName: '@next/swc-win32-arm64-msvc',
+      },
+    ],
+  },
+  linux: {
+    x64: [
+      {
+        packageName: '@next/swc-linux-x64-gnu',
+        check: () => isGlibc,
+      },
+      {
+        packageName: '@next/swc-linux-x64-musl',
+        check: () => !isGlibc,
+      },
+    ],
+    arm64: [
+      {
+        packageName: '@next/swc-linux-arm64-gnu',
+        check: () => isGlibc,
+      },
+      {
+        packageName: '@next/swc-linux-arm64-musl',
+        check: () => !isGlibc,
+      },
+    ],
+    arm: [
+      {
+        packageName: '@next/swc-linux-arm-gnueabihf',
+      },
+    ],
+  },
+  darwin: {
+    x64: [
+      {
+        packageName: '@next/swc-darwin-x64',
+      },
+    ],
+    arm64: [
+      {
+        packageName: '@next/swc-darwin-arm64',
+      },
+    ],
+  },
+  android: {
+    arm64: [
+      {
+        packageName: '@next/swc-android-arm64',
+      },
+    ],
+  },
+}
+
+let activePackage = packageMap[process.platform]?.[process.arch]?.find(
+  (item) => {
+    return typeof item.check === 'undefined' ? true : item.check()
+  }
+)
+
+if (!activePackage) {
+  // TODO: should this be a hard error even though it fails install?
+  // should we fallback to wasm in this case?
```

Comment:
Falling back to wasm makes sense to me.

Thats the advantage of this custom install script 👍

Reply:
Yeah, falling back to wasm sounds good to me.